### DISCUSSION TITLE
Defaul app size

### DISCUSSION THREAD

**Discussion by dimavs** [preactjs/preact-cli#1664] [General]
I've just run

$ npx preact-cli create default app
$ du -hs app
465M    app

Looks a bit crazy for a "fast 3kb react alternative" :-)

**Comment by rschristian** [comment]
Is this a joke?

A) Preact is a ~3kb minzipped lib, yes. This is an entire web application, with images, manifests, CSS, etc. Apples and oranges.

B) If you're just looking at raw directory size (and not even a build), that's including `node_modules` and all sorts of other files that would never be relevant to your bundle.

**Reply by dimavs** [reply]
Yeah, I understand. It was a joke, but only partially. More like a frustration after I run that create command. I've tried it again and this time put a time on it:
```
real    3m59.868s
user    1m2.602s
sys     0m8.328s
```

I would expect from _create_ just to create a new project from template and don't do any npm init or build/run/test/whatever. Like _dotnet new_ just creates a project, _dotnet restore_ downloads dependencies and _dotnet build/run/test_ just doing its job.

**Reply by rschristian** [reply]
What you expect differs from what others do, I suppose. I personally don't see the point of creating a new project and then _not installing_ the dependencies. What's the value in that? You don't have a working project until you manually change directory and run your install command. That's just wasted time, IMO.

Regardless, [if you take a look at our docs](https://github.com/preactjs/preact-cli#preact-create), you'd see this is just a CLI flag that's enabled by default. `npx preact-cli create default app --no-install` or `npx preact-cli create default app --install false` does what you'd want.

**Reply by rschristian** [reply]
There is a sort of issue with `npx` here, in that it has to install the full deps for `preact-cli` when `create` only uses a subset of these. There's no real nice way around this though without creating a separate module for project creation.
